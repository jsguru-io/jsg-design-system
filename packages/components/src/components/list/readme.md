# scale-list

<!-- Auto Generated Below -->


## Properties

| Property  | Attribute | Description                           | Type      | Default     |
| --------- | --------- | ------------------------------------- | --------- | ----------- |
| `ordered` | `ordered` | (optional) Make the list ordered (ol) | `boolean` | `false`     |
| `styles`  | `styles`  | (optional) Injected CSS styles        | `string`  | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
