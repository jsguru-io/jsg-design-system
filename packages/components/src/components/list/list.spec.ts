/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

import { newSpecPage } from '@stencil/core/testing';
import { List } from './list';

describe('List', () => {
  let element;
  beforeEach(async () => {
    element = new List();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [List],
      html: `<scale-list>default</scale-list>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.ordered = true;
    expect(element.getCssClassMap()).toContain('list--type-ordered');
  });

  it('should handle listItem', async () => {
    const page = await newSpecPage({
      components: [List],
      html: `
        <scale-list ordered>
          <scale-list-item>
          </scale-list-item>
        </scale-list>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
