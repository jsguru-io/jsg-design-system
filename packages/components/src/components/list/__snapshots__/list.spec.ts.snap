// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should handle listItem 1`] = `
<scale-list ordered="">
  <mock:shadow-root>
    <ol class="list list--type-ordered" part="base ordered">
      <slot></slot>
    </ol>
  </mock:shadow-root>
  <scale-list-item></scale-list-item>
</scale-list>
`;

exports[`List should match snapshot 1`] = `
<scale-list>
  <mock:shadow-root>
    <ul class="list" part="base">
      <slot></slot>
    </ul>
  </mock:shadow-root>
  default
</scale-list>
`;
